client/scss/components/forms/_input-text.scss

```diff
@@ -44,6 +44,6 @@ textarea {
 .w-field--date_time_field,
 .w-field--time_field {
   input {
-    width: auto;
+    max-width: 100%;
```

Comment:
Same as https://github.com/wagtail/wagtail/pull/12318#discussion_r1761315074,

This forces all date/time fields to always take up the full width of the parent, which is not what we want:

<img width="1168" alt="image" src="https://github.com/user-attachments/assets/d6905c78-af91-400b-91fe-fecbbd1e5080">

The issue's description specifically says **adding** `max-width`, not replacing the existing `width: auto`.